gpbft/rebroadcast.go

```diff
@@ -0,0 +1,151 @@
+package gpbft
+
+import (
+	"fmt"
+	"time"
+)
+
+type rebroadcastState struct {
+	timeout  time.Time
+	attempts int
+
+	decideCurrentRound,
+	commitCurrentRound,
+	commitPreviousRound,
+	prepareCurrentRound,
+	preparePreviousRound,
+	convergeCurrentRound *MessageBuilder
+
+	instanceAtLatestBroadcast uint64
+	roundAtLatestBroadcast    uint64
+	phaseAtLatestBroadcast    Phase
+}
+
+func (i *instance) tryRebroadcast() error {
+	if i.shouldRebroadcast() {
+		previousRebroadcast := i.participant.host.Time()
+		defer func() {
```

Comment:
Is this defer being used explicitly to ensure this logic runs even if the rebroadcast panics? If not, I think it would be much clearer to just call rebroadcast, remember the error value, and then return it later. And if we think the panics can happen, I reckon we should make a larger change to prevent them.

Reply:
> logic runs even if the rebroadcast panics

Yes that was my intention. It is totally up for discussion on how effective this strategy is, since if panics occur here we probably have bigger problems. Having said that I would like to avoid flurry of rebroadcasts (or total halt of rebroadcasts) if code does panic.

>  I reckon we should make a larger change to prevent them.

Yep that makes sense 👍